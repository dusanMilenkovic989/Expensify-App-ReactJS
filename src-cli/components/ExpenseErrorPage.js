import React from 'react'
import { Link } from 'react-router-dom'

const ExpenseErrorPage = () => (
    <div>
        <p>404 Page</p>
        <Link to="/dashboard">Go home</Link>
    </div>
)

export { ExpenseErrorPage as default }