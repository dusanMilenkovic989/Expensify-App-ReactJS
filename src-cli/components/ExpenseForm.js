import React from 'react'
import { SingleDatePicker } from 'react-dates'
import moment from 'moment'

class ExpenseForm extends React.Component {
    state = {
        description: this.props.expense ? this.props.expense.description : '',
        amount: this.props.expense ? (this.props.expense.amount / 100).toString() : '',
        note: this.props.expense ? this.props.expense.note : '',
        createdAt: this.props.expense ? moment(this.props.expense.createdAt) : moment(),
        calendarFocused: false,
        error: '' 
    }
    onDescriptionChange = (e) => {
        const description = e.target.value
        this.setState(() => ({ description }))
    }
    onAmountChange = (e) => {
        const amount = e.target.value

        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) this.setState(() => ({ amount }))
    }
    onDateChange = (createdAt) => {
        if (createdAt) this.setState(() => ({ createdAt }))
    }
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }))
    }
    onNoteChange = (e) => {
        const note = e.target.value
        this.setState(() => ({ note }))
    }
    onSubmit = (e) => {
        e.preventDefault()

        if (!this.state.description || !this.state.amount) return this.setState(() => ({ error: 'Description and amount must be provided!' }))

        this.setState(() => ({ error: '' }))
        this.props.onFormSubmit({
            description: this.state.description,
            amount: parseFloat(this.state.amount, 10) * 100,
            note: this.state.note,
            createdAt: this.state.createdAt.valueOf()
        })
    }
    render() {
        return (
            <form className="form" onSubmit={this.onSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <input
                    type="text"
                    placeholder="Description"
                    autoFocus
                    className="text-input"
                    value={this.state.description}
                    onChange={this.onDescriptionChange}
                />
                <input
                    type="text"
                    placeholder="Amount"
                    className="text-input"
                    value={this.state.amount}
                    onChange={this.onAmountChange}
                />
                <div>
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                </div>
                <textarea
                    placeholder="Add expense note here"
                    className="textarea"
                    value={this.state.note}
                    onChange={this.onNoteChange}
                />
                <div>
                    <button className="button">{this.props.expense ? 'Update Expense' : 'Add Expense'}</button>
                </div>
            </form>
        )
    }
}

export { ExpenseForm as default }