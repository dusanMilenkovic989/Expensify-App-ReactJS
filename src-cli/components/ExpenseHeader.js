import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import visibleExpenses from '../redux/selectors/expenses'
import summary from '../redux/selectors/summary'
import numeral from '../utils/numeral'

export const ExpenseHeader = ({ expenses }) => {
    const plural = expenses.length !== 1 ? 's' : ''
    const message = (
        <h1 className="page-header__title">
            Viewing <span>{expenses.length}</span> expense{plural}, totalling <span>{numeral(summary(expenses) / 100).format('0,0.00 $')}</span>
        </h1>
    )

    return (
        <div className="page-header">
            <div className="content-container">  
                {expenses.length > 0 && message}
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => ({ expenses: visibleExpenses(state.expenses, state.filters) })

export default connect(mapStateToProps)(ExpenseHeader)