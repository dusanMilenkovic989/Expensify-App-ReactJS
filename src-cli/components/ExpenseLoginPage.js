import React from 'react'
import { connect } from 'react-redux' 
import { login } from '../redux/actions/authentication'

export const ExpenseLoginPage = ({ login }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>Log in to start tracking your expenses</p>
            <button className="button" onClick={login}>Log in with Google</button>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({ login: () => { dispatch(login()) } })

export default connect(undefined, mapDispatchToProps)(ExpenseLoginPage)