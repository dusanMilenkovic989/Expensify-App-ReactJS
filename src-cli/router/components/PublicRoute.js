import React from 'react'
import { Navigate } from 'react-router-dom'
import { connect } from 'react-redux'

export const PublicRoute = ({ child, isAuthenticated }) => isAuthenticated ? ( <Navigate to="/dashboard" /> ) : child

const mapStateToProps = (state) => ({ isAuthenticated: !!state.authenticated.id })

export default connect(mapStateToProps)(PublicRoute)