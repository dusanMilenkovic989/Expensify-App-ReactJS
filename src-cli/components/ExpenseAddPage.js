import React from 'react'
import { useNavigate } from 'react-router-dom'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { addExpense } from '../redux/actions/expenses'

const withNavigateExpenseAddPage = (props) => {
    const navigate = useNavigate()

    return <ExpenseAddPage {...props} navigate={navigate} />
}

export const ExpenseAddPage = ({ addExpense, navigate }) => (
    <div>
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Add Expense</h1>
            </div>
        </div>
        <div className="content-container">
            <ExpenseForm 
                onFormSubmit={(expense) => { 
                    addExpense(expense)
                    navigate('/')
                }}
            />
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({ addExpense: (expense) => { dispatch(addExpense(expense)) } })

export default connect(undefined, mapDispatchToProps)(withNavigateExpenseAddPage)