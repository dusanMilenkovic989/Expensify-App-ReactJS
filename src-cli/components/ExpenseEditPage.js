import React from 'react'
import { useNavigate, useParams } from 'react-router-dom'
import { connect } from 'react-redux'
import { editExpense, removeExpense } from '../redux/actions/expenses'
import ExpenseForm from './ExpenseForm'

const withParAndNavExpenseEditPage = (props) => {
    const params = useParams()
    const navigate = useNavigate()

    return <ExpenseEditPage {...props} params={params} navigate={navigate} />
}

export const ExpenseEditPage = ({ expenses, editExpense, removeExpense, params, navigate }) => (
    <div>
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Edit Expense</h1>
            </div>
        </div>
        <div className="content-container">
            <ExpenseForm 
                onFormSubmit={(expense) => { 
                    editExpense(params.id, expense)
                    navigate('/')
                }}
                expense={expenses.find((expense) => expense.id === params.id)}
            />
            <button
                className="button button--secondary"
                onClick={() => { 
                    removeExpense({ id: params.id })
                    navigate('/')
                }}
            >
                Remove expense
            </button>
        </div>
    </div>
)

const mapStateToProps = (state) => ({ expenses: state.expenses })

const mapDispatchToProps = (dispatch) => ({ 
    editExpense: (expenseId, expense) => { dispatch(editExpense(expenseId, expense)) },
    removeExpense: (data) => { dispatch(removeExpense(data)) }
})

export default connect(mapStateToProps, mapDispatchToProps)(withParAndNavExpenseEditPage)