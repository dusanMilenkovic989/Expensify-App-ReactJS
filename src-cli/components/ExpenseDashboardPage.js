import React from 'react'
import ExpenseHeader from './ExpenseHeader'
import ExpenseListFilters from './ExpenseListFilters'
import ExpenseList from './ExpenseList'

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseHeader />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

export { ExpenseDashboardPage as default }