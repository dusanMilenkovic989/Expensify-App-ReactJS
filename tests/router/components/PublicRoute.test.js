import React from 'react'
import { shallow } from 'enzyme'
import { PublicRoute } from '../../../src-cli/router/components/PublicRoute'
import { ExpenseLoginPage } from '../../../src-cli/components/ExpenseLoginPage'

test('Should render PublicRoute correctly when user is authenticated', () => {
    const wrapper = shallow(<PublicRoute child={<ExpenseLoginPage />} isAuthenticated={true} />)

    expect(wrapper).toMatchSnapshot()
})

test('Should render PublicRoute correctly when user is not authenticated', () => {
    const wrapper = shallow(<PublicRoute child={<ExpenseLoginPage />} isAuthenticated={false} />)

    expect(wrapper).toMatchSnapshot()
})