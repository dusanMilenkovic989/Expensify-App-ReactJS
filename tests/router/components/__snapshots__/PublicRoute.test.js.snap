// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PublicRoute correctly when user is authenticated 1`] = `
<Navigate
  to="/dashboard"
/>
`;

exports[`Should render PublicRoute correctly when user is not authenticated 1`] = `<ExpenseLoginPage />`;
